### PR TITLE
fix(communitypermissions): fixed dropdown height

### PR DESCRIPTION
**Pull request by alexandraB99**
### What does the PR do
community permissions, "InDropDown" fixed dropdown height

### Affected areas
community permissions, "InDropDown"

### Screenshot of functionality (including design for comparison)
https://user-images.githubusercontent.com/31625338/218122102-ff6dbe0d-edcc-40fa-b9df-af22d90ad026.mov

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Popups/StatusMenuSeparator.qml [hunk 1]
```diff
@@ -4,7 +4,7 @@ import QtQuick.Controls 2.13
 import StatusQ.Core.Theme 0.1
 
 MenuSeparator {
-    height: visible && enabled ? implicitHeight : 0
+    implicitHeight: visible && enabled ? implicitHeight : 0
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Popups/StatusMenuSeparator.qml:7]: Why this change?

Comment (alexandraB99) [reply ui/StatusQ/src/StatusQ/Popups/StatusMenuSeparator.qml:7]: ah, because is used in Layout type

Comment (alexandraB99) [reply ui/StatusQ/src/StatusQ/Popups/StatusMenuSeparator.qml:7]: although now that I see it probably it's not necessary

Comment (alexandraB99) [reply ui/StatusQ/src/StatusQ/Popups/StatusMenuSeparator.qml]: removed! also fixed id verification popup 
pp

**Review by caybro — APPROVED** [review]

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml [hunk 2]
```diff
@@ -154,12 +148,12 @@ StatusDropdown {
...
         }
-
         StatusScrollView {
             id: scrollView
-
             Layout.fillWidth: true
-            Layout.fillHeight: true
+            Layout.minimumHeight: (((topRepeater.count > 5) ? 5 : topRepeater.count)*44)
+            Layout.maximumHeight: (((topRepeater.count > 5) ? 5 : topRepeater.count)*44)
+            contentHeight: scollableColumn.implicitHeight
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:156]:
```suggestion
            contentHeight: scrollableColumn.implicitHeight
```

Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:156]:
```suggestion
            contentHeight: scrollableColumn.implicitHeight
```

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml [hunk 3]
```diff
@@ -154,12 +148,12 @@ StatusDropdown {
 
             visible: root.allowChoosingEntireCommunity
         }
-
         StatusScrollView {
             id: scrollView
-
             Layout.fillWidth: true
-            Layout.fillHeight: true
+            Layout.minimumHeight: (((topRepeater.count > 5) ? 5 : topRepeater.count)*44)
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:154]: Since 44 is the same value than list item `Layout.preferredHeight` could we use a `d` property instead and share accross the needed properites?

Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:154]: Same with the 5 value. Could we use it as `d` property if it is a `default` margin?

**Review by noeliaSD — COMMENTED** [review]
I tested it and looks good.

Just added some minor comments!

**Comment by alexandraB99** [comment]
thanks for you suggestions @noeliaSD, all done!

**Comment by noeliaSD** [comment]
> thanks for you suggestions @noeliaSD, all done!

Thanks @alexandraB99 for the updates! They look good!! :)

Just 2 other more things:
- I see now a new `d` property `defaultVMargin = 9` but it is already not used, right?
- Is your intention to use it to replace the `5` number that is used in lines 156 / 157? If not, then, it would be also good to add another property for this value!

**Comment by alexandraB99** [comment]
> > thanks for you suggestions @noeliaSD, all done!
> 
> Thanks @alexandraB99 for the updates! They look good!! :)
> 
> Just 2 other more things:
> 
>     * I see now a new `d` property `defaultVMargin = 9` but it is already not used, right?
> 
>     * Is your intention to use it to replace the `5` number that is used in lines 156 / 157? If not, then, it would be also good to add another property for this value!

all should be fine now :D

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml [hunk 4]
```diff
@@ -154,27 +151,26 @@ StatusDropdown {
...
             visible: root.allowChoosingEntireCommunity
         }
-
         StatusScrollView {
             id: scrollView
-
             Layout.fillWidth: true
-            Layout.fillHeight: true
-            Layout.bottomMargin: 9
+            Layout.minimumHeight: (((topRepeater.count > d.maxHeightCountNo) ? d.maxHeightCountNo : topRepeater.count)*d.itemStandardHeight)
```
Comment (micieslak) [inline ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:157]: It would be simpler and more readable imo:
```
Layout.minimumHeight: Math.min(d.maxHeightCountNo, topRepeater.count) * d.itemStandardHeight
```

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml [hunk 5]
```diff
@@ -154,27 +151,26 @@ StatusDropdown {
...
+            Layout.maximumHeight: (((topRepeater.count > d.maxHeightCountNo) ? d.maxHeightCountNo : topRepeater.count)*d.itemStandardHeight)
```
Comment (micieslak) [inline ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:158]: To avoid repetition it could be just
```
Layout.maximumHeight: Layout.minimumHeight
```

**Review by micieslak — CHANGES_REQUESTED** [review]
It looks that there is a problem with width of the dropdown content:

Screenshot from 2023-02-16 00-41-09

checkboxes seems to be cut off, design for comparison:

Screenshot from 2023-02-16 00-42-38

Figma link: https://www.figma.com/file/17fc13UBFvInrLgNUKJJg5/Kuba%E2%8E%9CDesktop?node-id=2934%3A478959&t=dbRjJ3lifuiVNFQu-0

**Comment by micieslak** [comment]
The strange thing is that checkboxes are visible in the app but not in storybook. It maybe related to changes in channels list structure done recently (model flattened). Bc of that categories are not displayed properly in the app (we need a ticket for that - will create later).

**Review by noeliaSD — APPROVED** [review]
LGTM, apart from the issue commented by Michal!

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml (hunk 4 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Chat/controls/community/InDropdown.qml]: definitely!

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml (hunk 5 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Chat/controls/community/InDropdown.qml]: done

**Comment by alexandraB99** [comment]
> The strange thing is that checkboxes are visible in the app but not in storybook. It maybe related to changes in channels list structure done recently (model flattened). Bc of that categories are not displayed properly in the app (we need a ticket for that - will create later).

@micieslak I can see all check boxes in the app, I cannot see the page in storybook though, I get:
AppLayouts/Chat/stores/CommunitiesStore.qml:172: TypeError: Cannot read property 'tokensModel' of undefined
AppLayouts/Chat/stores/CommunitiesStore.qml:8: ReferenceError: mintingModule is not defined

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml [hunk 6]
```diff
@@ -203,8 +200,7 @@ StatusDropdown {
 
                     ColumnLayout {
                         id: column
-
-                        Layout.fillWidth: true
+                        width: parent.width
```
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/app/AppLayouts/Chat/controls/community/InDropdown.qml:203]: There is also another problem, but I created a separate ticket for that: https://github.com/status-im/status-desktop/issues/9591 `InDropdown` needs to be adapted to changes done by @jrainville (linked in the ticket).

**Review by micieslak — CHANGES_REQUESTED** [review]

File: ui/app/AppLayouts/Chat/controls/community/InDropdown.qml (hunk 6 above)
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Chat/controls/community/InDropdown.qml]: @micieslak fixed it already before I saw your comments, please have another look and let me know if any other comments

**Review by micieslak — APPROVED** [review]
lgtm!

**Review by caybro — CHANGES_REQUESTED** [review]

File: ui/imports/shared/popups/OutgoingContactVerificationRequestPopup.qml [hunk 7]
```diff
@@ -66,17 +66,21 @@ StatusDialog {
...
+        implicitHeight: Math.min(width,contentHeight)
+        contentWidth: width
+        contentHeight: columnCtx.height
+        anchors.horizontalCenter: parent.horizontalCenter
+        padding: 0
+        Column {
+            id: columnCtx
+            anchors.left: parent.left
+            anchors.right: parent.right
+            anchors.margins: Style.current.bigPadding
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/imports/shared/popups/OutgoingContactVerificationRequestPopup.qml:79]: And even if it's rebased, it's already correct. How is this related to `fix(communitypermissions): fixed dropdown height`?

**Review by caybro — APPROVED** [review]
LGTM
